src/coreclr/tools/Common/TypeSystem/Ecma/SymbolReader/UnmanagedPdbSymbolReader.cs

```diff
@@ -82,17 +229,231 @@ private interface IMetaDataDispenser
             // Don't need any other methods.
         }
 
+#if NET6_0_OR_GREATER
+        private sealed class MetaDataDispenserWrapperCache : ComWrappers
+        {
+            public static readonly MetaDataDispenserWrapperCache Instance = new MetaDataDispenserWrapperCache();
+            private MetaDataDispenserWrapperCache() { }
+
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count) => throw new NotImplementedException();
+            protected override object CreateObject(IntPtr externalComObject, CreateObjectFlags flags)
+            {
+                Debug.Assert(flags == CreateObjectFlags.UniqueInstance);
+
+                return new MetaDataDispenserRcw(externalComObject);
+            }
+            protected override void ReleaseObjects(IEnumerable objects) => throw new NotImplementedException();
+
+            public sealed unsafe record MetaDataDispenserRcw(IntPtr Inst) : IMetaDataDispenser
+            {
+                private bool _disposed = false;
+
+                public void DefineScope_Placeholder() => throw new NotImplementedException();
+
+                /// <remarks>
+                /// <paramref="punk" /> is simply a boxed IntPtr, because we don't need an RCW.
+                /// </remarks>
+                public int OpenScope(string szScope, int dwOpenFlags, ref Guid riid, out object punk)
+                {
+                    var func = (delegate* unmanaged<IntPtr, char*, int, Guid*, IntPtr*, int>)(*(*(void***)Inst + 4));
+                    IntPtr outPtr;
+                    int hr;
+                    fixed (char* szScopePtr = szScope)
+                    fixed (Guid* riidPtr = &riid)
+                    {
+                        hr = func(Inst, szScopePtr, dwOpenFlags, riidPtr, &outPtr);
+                    }
+                    punk = hr == 0 ? (object)outPtr : null;
```

Comment:
I'd create a dummy RCW object for the `outPtr` here and then check for the type elsewhere. We are already boxing and allocating a type, let's do it in a manner that is nicely typed rather than boxing.